.github/workflows/cd.yml

```diff
@@ -1,13 +1,11 @@
 name: cd
+
 on:
-  workflow_run:
-    workflows: ["ci"]
-    branches-ignore: [master]
-    types:
-      - completed
-  push:
-    tags:
-      - "v[0-9]+.[0-9]+.[0-9]+"
+  workflow_call:
```

Comment:
you still need the `workflow_run` but instead for `branches-ignore` add `["*"]`.

Reply:
`workflow_run` always works on default branch. So It will `build` & `push` for default branch.